source/MaterialXTest/Runtime.cpp

```diff
@@ -1428,6 +1428,22 @@ TEST_CASE("Runtime: materials", "[runtime]")
     api->loadLibrary(STDLIB);
     api->loadLibrary(PBRLIB);
     api->loadLibrary(BXDFLIB);
+
+    // Set and test search paths
+    api->clearSearchPath();
+    REQUIRE(api->getSearchPath().isEmpty());
+    api->setSearchPath(searchPath);
+    REQUIRE(api->getSearchPath().asString() == searchPath.asString());
+
+    REQUIRE(api->getTextureSearchPath().isEmpty());
+    mx::FileSearchPath texturePath(mx::FilePath::getCurrentPath() / mx::FilePath("resources/Images"));
+    api->setTextureSearchPath(texturePath);
+    REQUIRE(api->getTextureSearchPath().find("brass_color.jpg").exists());
+
+    REQUIRE(api->getImplementationSearchPath().isEmpty());
+    mx::FileSearchPath implPath(mx::FilePath::getCurrentPath() / mx::FilePath("libraries/stdlib/genglsl"));
+    api->setImplementationSearchPath(implPath);
+    REQUIRE(api->getImplementationSearchPath().find("stdlib_genglsl_unit_impl.mtlx").exists());    
```

Comment:
Note: This is not a recursive search. @feldstj, as mentioned we'll need to use or port over the recursive search previously implemented.